samtranslator/translator/translator.py

```diff
@@ -269,6 +272,64 @@ def _get_resources_to_iterate(
         return functions + statemachines + apis + others + connectors
 
 
+def make_error(msg: str) -> Exception:
+    return InvalidDocumentException([InvalidTemplateException(msg)])
+
+
+def add_embedded_connectors(sam_template: Dict[str, Any]) -> Any:
+    resources = sam_template.get("Resources", {})
+    connectors_list = []
+
+    # Loop through the resources in the template and see if any connectors have been attached
+    for source_logicalId, resource in resources.items():
+        if "Connectors" in resource:
+            if not isinstance(resource["Connectors"], dict):
+                raise make_error("'Connectors' must be a dict")
+
+            # Go through each of the connectors that have been attached and create a Serverless Connector resource
+            for connector_logicalId, connector_dict in resource["Connectors"].items():
```

Comment:
Does this `connector_logicalId` needs to be unique in the whole template? It seems a bit hard to come up with different names for customers.

Reply:
Yea it has to be unique. We decided to allow users to set the logical id for the generated connectors to take away any convoluted logical id generation and to make it easier for them to reference the connectors if they needed to.